lib/lightning_web/live/project_live/settings.ex

```diff
@@ -28,6 +27,7 @@ defmodule LightningWeb.ProjectLive.Settings do
   on_mount {LightningWeb.Hooks, :project_scope}
   on_mount {LightningWeb.Hooks, :limit_github_sync}
   on_mount {LightningWeb.Hooks, :limit_mfa}
+  on_mount {LightningWeb.Hooks, :limit_retention_periods}
```

Comment:
Thinking out loud here, I think these hooks are needing a rename, afaik all the hooks in the file are related to projects.

Also, at least 2/3 of these are only used in this live view, so we might be extending an 'implicit contract' - where this one live view expects a bunch of assigns that are added by another module.

I don't necessarily thing we need to change this now - but the settings module (when using hooks) feels a bit 'stringy' where we quietly plug in a bunch of assigns that are all very specific.

Reply:
Yeah, good point. Feels like we need a central place to extend the settings live view with limiters and extensions